src/prog_models/utils/containers.py

```diff
@@ -35,66 +35,91 @@ def __init__(self, keys : list, data : Union[dict, np.array]):
                     [data[key]] if key in data else [None] for key in keys
                 ], dtype=np.float64)
         else:
-            raise ProgModelTypeError(f"Data must be a dictionary or numpy array, not {type(data)}")     
+            raise ProgModelTypeError(f"Data must be a dictionary or numpy array, not {type(data)}")
 
     def __reduce__(self):
         return (DictLikeMatrixWrapper, (self._keys, self.matrix))
 
-    def __getitem__(self, key : str) -> int:
-        row = self.matrix[self._keys.index(key)]
-        if len(row) == 1:
-            return self.matrix[self._keys.index(key)][0]
-        return self.matrix[self._keys.index(key)]
+    # get all values associated with a key, ex: all values of 'i'
```

Comment:
Could we move these comments that are above the function declaration to doc comments (three quotes) after the function declaration)? That way they'll be included in hints for many IDEs and generated documentation.

This applies to all the others below as well.